libraries/AP_RangeFinder/AP_RangeFinder_RDS02UF.cpp

```diff
@@ -0,0 +1,225 @@
+            }
+            break;
+        }
+        case RDS02UF_PARSE_STATE::STATE8_REAL_DATA: // real_data
+            parser_buffer[parser_index++] = data;
+            if (parser_index == (RDS02UF_HEAD_LEN + RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN) ) {
+                decode_state = RDS02UF_PARSE_STATE::STATE9_CRC;
+            }
+            break;
+        case RDS02UF_PARSE_STATE::STATE9_CRC: // crc
+        {
+        #if RDS02_USE_CRC
+	        const uint8_t crc_data = crc8(&parser_buffer[2], RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN);
+            if (data == crc_data || data == RDS02UF_IGNORE_CRC) { // 0xff indicates that the current frame does not need to be checked.
+                parser_buffer[parser_index++] = data;
+                decode_state = RDS02UF_PARSE_STATE::STATE10_END_1;
+            } else {
+                parser_index = 0;
+                decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+            }
+        #else
+            parser_buffer[parser_index++] = data;
+            decode_state = RDS02UF_PARSE_STATE::STATE10_END_1;
+        #endif
+            break;
+        }
+        case RDS02UF_PARSE_STATE::STATE10_END_1:
+            if (data == RDS02_END) {
+                parser_buffer[parser_index++] = data;
+                decode_state = RDS02UF_PARSE_STATE::STATE11_END_2;
+            } else {
+                parser_index = 0;
+                decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+            }
+            break;
+        case RDS02UF_PARSE_STATE::STATE11_END_2:
+        {
+            uint16_t fc_code = (parser_buffer[RDS02UF_PRODUCTS_FC_INDEX_H] << 8 | parser_buffer[RDS02UF_PRODUCTS_FC_INDEX_L]);
+            uint8_t err_code = parser_buffer[RDS02UF_ERROR_CODE_INDEX];
+
+            if (data == RDS02_END) {
+                if (fc_code == RDS02UF_UAV_PRODUCTS_ID && err_code == RDS02UF_NO_ERR) {// get targer information
+                    uint16_t read_info_fc = (parser_buffer[RDS02_TARGET_FC_INDEX_H] << 8 | parser_buffer[RDS02_TARGET_FC_INDEX_L]);
+                    if ((read_info_fc & RDS02UF_IGNORE_ID_BYTE) == RDS02_TARGET_INFO_FC){	// read_info_fc = 0x70C + ID * 0x10, ID: 0~0xF
+                        _distance_m = (parser_buffer[RDS02_DATA_Y_INDEX_H] * 256 + parser_buffer[RDS02_DATA_Y_INDEX_L]) / 100.0f;
+                        if (_distance_m > RDS02UF_DIST_MAX_M) {
+                            _distance_m = RDS02UF_DIST_MAX_M;
+                            set_status(RangeFinder::Status::OutOfRangeHigh);
+                        } else if (_distance_m < RDS02UF_DIST_MIN_M) {
+                            _distance_m = RDS02UF_DIST_MIN_M;
+                            set_status(RangeFinder::Status::OutOfRangeLow);
+                        }
+                        ret = true;
+                        state.last_reading_ms = AP_HAL::millis();
+                    }
+
+                }
+            }
+			parser_index = 0;
```

Comment:
ardupilot indentation is whitespace 4.

Reply:
I didn't notice that tabs are used here, I will make a modification, thank you for pointing out the problem